libpod/boltdb_state.go

```diff
@@ -1472,7 +1473,17 @@ func (s *BoltState) PruneContainerExitCodes() error {
 			return err
 		}
 
+		ctrsBucket, err := getCtrBucket(tx)
+		if err != nil {
+			return err
+		}
+
 		return timeStampBucket.ForEach(func(rawID, rawTimeStamp []byte) error {
+			if ctrsBucket.Bucket(rawID) != nil {
+				// If the container still exists, don't prune
+				// its exit code since we may still need it.
+				return nil
+			}
```

Comment:
This seems like a bad idea? This could cause us to grow this table infinite. Consider the common use case where people just do podman start/stop and not recreate the container every time.

Reply:
I don't think it can grow indefinitely.  There will always only be 1 exit code per container, it's not a growing list or something like that.

The idea here is to just not delete this 1 exit code prematurely (since we need it for the exit-code policies to work).

Reply:
oh I see, I thought the db was keyed `ts -> code` but it is `id -> code` and `id -> ts` so we can only ever store one. In this case this is fine.